hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/LoadBalancingKMSClientProvider.java

```diff
@@ -233,6 +233,10 @@ private LoadBalancingKMSClientProvider(URI uri,
   }
 
   private int nextIdx() {
+    if (providers.length == 0) {
```

Comment:
Thanks @yiyuaner  for reporting the issue. Should we check providers.length in constructor  as there are many other places assumes providers.length is not 0.

Reply:
Ok. I have moved the checks to the constructor.